src/tests/mock/http.rs

```diff
@@ -255,3 +303,196 @@ fn full_path(path: Option<String>, query: Option<String>) -> String {
+                            {
+                                current_path.pop();
+                                return Err(e);
+                            }
+                        }
+                        None => {
+                            let error_msg = format!(
+                                "Key missing in expected JSON at {}: {:?}",
+                                current_path.join("."),
+                                key
+                            );
+                            current_path.pop();
+                            return Err(anyhow!(error_msg));
+                        }
+                    }
+
+                    current_path.pop();
+                }
+                Ok(())
+            }
+            _ => {
+                if actual != expected {
+                    Err(anyhow!(
+                        "Mismatch at {}: {:?} != {:?}",
+                        current_path.join("."),
+                        actual,
+                        expected
+                    ))
+                } else {
+                    Ok(())
+                }
+            }
+        }
+    }
+
+    let mut current_path = vec![];
+    compare_jsons(actual, expected, exclude_paths, &mut current_path)
+}
+
+#[cfg(test)]
+mod tests {
+    use {super::*, maplit::hashset, serde_json::json};
+
+    #[test]
+    fn test_parse_field_paths() {
+        let paths = ["user.profile.name", "user.settings"];
+        let parsed_paths = parse_field_paths(&paths);
+        let expected_paths: HashSet<Vec<String>> = hashset! {
+            vec!["user".to_string(), "profile".to_string(), "name".to_string()],
+            vec!["user".to_string(), "settings".to_string()],
+        };
+        assert_eq!(parsed_paths, expected_paths)
+    }
+
+    #[test]
+    fn test_json_matches_excluding_no_exclusions() {
+        let json_a = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice"
+                }
+            }
+        });
+        let json_b = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice"
+                }
+            }
+        });
+        assert_json_matches!(json_a, json_b, [])
+    }
+
+    #[test]
+    fn test_json_matches_excluding_with_exclusions() {
+        let json_a = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice",
+                    "timestamp": "2021-01-01T12:00:00Z"
+                }
+            }
+        });
+        let json_b = json!({
+            "user": {
+                "id": 123,
+                "profile": {
+                    "name": "Alice",
+                    "timestamp": "2022-01-01T12:00:00Z"
+                }
+            }
+        });
+        assert_json_matches!(json_a, json_b, ["user.profile.timestamp"])
```

Comment:
Please, could you add a test with more than one element in the array? :pray:

Reply:
Added